include/ccf/http_etag.h

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#define FMT_HEADER_ONLY
+#include <regex>
+#include <set>
+#include <string>
+
+namespace ccf
+{
+  namespace http
+  {
+    /** Utility class to resolve If-Match and If-None-Match as described
+     * in https://www.rfc-editor.org/rfc/rfc9110#field.if-match
+     */
+    class Matcher
+    {
+    private:
+      /// If-Match header is not present
+      bool _empty = false;
+      /// If-Match header is present and has the value "*"
+      bool any_value = false;
+      /// If-Match header is present and has specific etag values
+      std::set<std::string> if_etags;
+
+    public:
+      /** Construct a Matcher from a match header if present
+       *
+       * Note: Weak tags are not supported.
+       */
+      Matcher(const std::optional<std::string>& match_header)
+      {
+        if (!match_header.has_value())
+        {
+          _empty = true;
+          return;
+        }
+
+        if (match_header.value() == "*")
+        {
+          any_value = true;
+          return;
+        }
+
+        std::regex etag_rx("\\\"([0-9a-f]+)\\\",?\\s*");
```

Comment:
Trying to understand if this regex is necessary, or overkill. Could we get the same result with:
```
etag_values = nonstd::split(match_header.value(), ",")
foreach tag_value in etag_values:
  Strip whitespace
  Check it starts and ends with "
  Append to if_etags
```

This would be more readable than a (heavily-escaped) regex, if I'm not missing anything.

Reply:
We probably could, I started down this road because I was hoping to capture all the values in one match(), but the API is awful and that's just not possible. The rest is path dependent.